app/controllers/api/v1/journeys_controller.rb

```diff
@@ -78,9 +78,10 @@ def new_journey_params
 
       def new_journey_attributes
         @new_journey_attributes ||= new_journey_params.to_h[:attributes].tap do |attribs|
+          attribs.delete(:timestamp) # throw the timestamp away as we are using client_timestamp instead
```

Comment:
Something about this change isn't feeling right to me. In particular the change to call `timestamp` method below suggests use of `data_params` rather than the `new_journey_params` hash - this is confusing and might cause side effects. If this is the intent, perhaps this change makes the behaviour more explicit:

```ruby
timestamp = attribs.delete(:timestamp)

attribs.merge!(
  ...
  client_timestamp: Time.zone.parse(timezone)
  ...
)
```

Reply:
happy to change as you suggest. NB further on in this file (line 135) timestamp is used again to create the event, and so it will be reparsed